element/daterange/classes/element.php

```diff
@@ -262,15 +268,22 @@ public function validate_form_elements($data, $files) {
             }
         }
 
-        // Check that date is correctly set.
         for ($i = 0; $i < $data['numranges']; $i++) {
             $enabled = $this->build_element_name('enabled', $i);
+            $recurring = $this->build_element_name('recurring', $i);
             $startdate = $this->build_element_name('startdate', $i);
             $enddate = $this->build_element_name('enddate', $i);
+            $rangeperiod = $data[$enddate] - $data[$startdate];
 
+            // Check that end date is correctly set.
             if (!empty($data[$enabled]) && $data[$startdate] >= $data[$enddate] ) {
                 $errors[$this->build_element_name('enddate', $i)] = get_string('error:enddate', 'customcertelement_daterange');
             }
+
+            // Check that recurring dateranges are not longer than 12 months.
+            if (!empty($data[$recurring]) && $rangeperiod > self::MAX_RECURRING_PERIOD ) {
```

Comment:
I think this should be >= here instead of >.
Otherwise, I would be able to define a daterange from for example 1st october 2017 to 1st october 2018 which is in fact more than the 12 months limit